ts/features/pn/hooks/__tests__/usePnPreferencesFetcher.test.tsx

```diff
@@ -0,0 +1,139 @@
+}));
+
+type PreferencePotState = pot.Pot<
+  ServicePreferenceResponse,
+  WithServiceID<NetworkError>
+>;
+
+// eslint-disable-next-line functional/no-let
+let testingHookData: {
+  isLoading: boolean;
+  isError: boolean;
+  isEnabled: boolean;
+};
+
+const mockDispatch = jest.fn();
+const mockUseIODispatch = useIODispatch as jest.Mock;
+const mockServicePreferencePotSelector =
+  servicePreferencePotSelector as jest.Mock;
+
+const pnServiceId = "PN_SID" as ServiceId;
+describe("usePnPreferencesFetcher", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockUseIODispatch.mockReturnValue(mockDispatch);
+  });
+
+  describe("should return correct values", () => {
+    const expectedReturnTypes = [
+      { isLoading: true, isError: false, isEnabled: true },
+      { isLoading: false, isError: false, isEnabled: true },
+      { isLoading: false, isError: false, isEnabled: false },
+      { isLoading: false, isError: true, isEnabled: true }
+    ];
+
+    const preferenceCases = [
+      pot.someLoading({
+        id: pnServiceId,
+        kind: "success",
+        value: { inbox: true }
+      }),
+      pot.some({
+        id: pnServiceId,
+        kind: "success",
+        value: { inbox: true }
+      }),
+      pot.some({
+        id: pnServiceId,
+        kind: "success",
+        value: { inbox: false }
+      }),
+      pot.someError(
+        {
+          id: pnServiceId,
+          kind: "success",
+          value: { inbox: true }
+        },
+        { id: pnServiceId, kind: "error", value: new Error("whatever") }
+      )
+    ] as Array<PreferencePotState>;
+    const titles = [
+      "pot.someLoading",
+      "pot.some(true)",
+      "pot.some(false)",
+      "error"
+    ];
+
+    for (const [index, title] of titles.entries()) {
+      it(` when id is correct and preference is ${title}`, () => {
+        const servicePreferencePot = preferenceCases[+index];
+
+        mockServicePreferencePotSelector.mockReturnValue(servicePreferencePot);
+
+        renderHook();
+
+        expect(testingHookData).toEqual(expectedReturnTypes[+index]);
+      });
+    }
+  });
+
+  it("should dispatch loadServicePreference, and return a loading state if service data is not already loaded", () => {
+    jest.spyOn(React, "useState").mockImplementation(() => [false, jest.fn()]);
+    const servicePreferencePot = pot.none;
+
+    mockServicePreferencePotSelector.mockReturnValue(servicePreferencePot);
+
+    renderHook();
+
+    expect(testingHookData.isLoading).toBe(true);
+    expect(testingHookData.isError).toBe(false);
+    expect(testingHookData.isEnabled).toBe(false);
+    expect(mockDispatch).toHaveBeenCalledWith(
+      loadServicePreference.request(pnServiceId)
+    );
+  });
+});
+
+const renderHook = () => {
+  const Component = () => {
+    const data = usePnPreferencesFetcher(pnServiceId);
+    testingHookData = data;
+    return <></>;
+  };
+  const globalState = appReducer(
+    {} as GlobalState,
```

Comment:
Why not undefined so it automatically gets initialized with default initial state values?

Reply:
sure!